### DISCUSSION TITLE
Calling one powershell notebook from another

### DISCUSSION THREAD

**Discussion by sudhagart** [dotnet/interactive#1285] [Q&A]
Hello folks,
  I would like to mark the first cell of one of my powershell notebooks as parameters and then invoke that notebook from another notebook using various values for the parameters. 

   jupyter supports this using the %run magic command or using papermill extensions where you can invoke the notebook with various options via papermill. 

   Is this supported with .Net interactive notebooks especially with powershell kernel?
   Are there other ways of accomplishing this workflow? 

Thanks

**Comment by jonsequitur** [comment]
.NET Interactive will work in Papermill. 

If you don't want to have to go through Jupyter and Python, @dfinke has also done a bunch of great work on this. Take a look: https://github.com/dfinke/PowerShellNotebook.

We're also investigating approaches that could work across multiple languages supported by .NET Interactive. One of the open questions is how to identify parameters. You can see some related issues under the [Automation](https://github.com/dotnet/interactive/labels/Area-Automation) label.

**Reply by dfinke** [reply]
To update that, my `Invoke-ExecuteNotebook` that runs notebooks headless and can inject and set parameters, works only with the native PowerShell where you start `Invoke-ExecuteNotebook`. I have not figured out how to load the .net interactive DLLs and use that PS runspace, plus if you use something like Plotly, it also won't render the graph.

On the Papermill side, I tried it, but it did not support parameters for PowerShell. I tried it a few weeks back, it is possible I did not do it correctly.

**Reply by jonsequitur** [reply]
Thanks @dfinke! Sorry about Papermill. I knew there was a community PR a while back to support .NET Interactive but I haven't tried it myself. 

@sudhagart, in the notebook containing the parameters, how would you expect to provide values for them when you open that notebook directly? The Papermill reliance on a specific frontend extension for parameter UI is something we're hoping to avoid, but we haven't yet designed a way to identify which variables are parameterizable, especially considering the multi-language notebook scenario.

**Reply by dfinke** [reply]
@jonsequitur No problem on the Papermill. I only tried it on a notebook with PS cells. It might work with others, and it may work straight away if you don't supply parameters.

**Reply by sudhagart** [reply]
Thank you for the responses so far. 

My usecase is to build a standalone powershell notebook with default parameters for troubleshooting workflow. And then have the notebook be invoked in other places with custom parameters potentially.

I did experiment with papermill module installed on python. It works calling powershell notebook from a python notebook without parameter passing. Parameter passing requires a papermill translator for the kernel to be built for powershell. 

https://papermill.readthedocs.io/en/latest/reference/papermill-translators.html 

Thanks 
Sudha

**Reply by sudhagart** [reply]
I am able to workaround this using @dfinke's powershell module for now. Digging more into that.. Thanks all for the answers.

**Reply by dfinke** [reply]
Glad that's working for you @sudhagart. Just to caution again, that headless approach uses the PowerShell runspace in "native" PowerShell, not the .NET interactive kernel. If your notebook uses magic commands or other .NET interactive kernel specifics, `Invoke-ExecuteNotebook` will fail.
